src/components/SearchAndSort/SearchAndSort.test.js

```diff
@@ -0,0 +1,359 @@
+        description: 'Test Description 1',
+        id: '80898dee-449f-44dd-9c8e-37d5eb469b1d',
+        name: 'Test Name 1',
+        updated: '3/16/2021',
+        updatedBy: 'System',
+      },
+    ],
+    resource: 'jobProfiles',
+  },
+});
+
+const mutator = buildMutator({
+  query: {
+    replace: noop,
+    update: noop,
+  },
+  resultCount: { replace: noop },
+});
+
+const searchAndSortProps = ({
+  parentResources,
+  isFullScreen,
+  layer,
+}) => ({
+  history: getHistory(layer),
+  objectName: 'testObjectName',
+  resultsLabel: <span>Results Label</span>,
+  searchLabelKey: 'ui-data-import.settings.jobProfiles.title',
+  resultCountMessageKey: 'ui-data-import.settings.jobProfiles.count',
+  fullWidthContainer: <span>fullWidthContainer</span>,
+  finishedResourceName: 'jobProfiles',
+  actionMenu: jest.fn(() => <span>Add</span>),
+  isFullScreen,
+  visibleColumns: [
+    'name',
+    'description',
+    'updated',
+    'updatedBy',
+  ],
+  nsParams: 'sort',
+  parentResources,
+});
+
+const getSearchAndSortComponent = ({
+  objectName,
+  resultsLabel,
+  searchLabelKey,
+  resultCountMessageKey,
+  finishedResourceName,
+  actionMenu,
+  isFullScreen,
+  fullWidthContainer,
+  parentResources,
+  visibleColumns,
+  nsParams,
+  history,
+}) => (
+  <Harness translations={translationsProperties}>
+    <Router history={history}>
+      <SearchAndSort
+        stripes={stripes}
+        objectName={objectName}
+        resultsLabel={resultsLabel}
+        initialResultCount={100}
+        resultCountIncrement={10}
+        searchLabelKey={searchLabelKey}
+        resultCountMessageKey={resultCountMessageKey}
+        showSingleResult
+        fullWidthContainer={fullWidthContainer}
+        match={{ path: '' }}
+        parentMutator={mutator}
+        parentResources={parentResources}
+        ViewRecordComponent={noop}
+        finishedResourceName={finishedResourceName}
+        actionMenu={actionMenu}
+        isFullScreen={isFullScreen}
+        EditRecordComponent={EditRecordComponentMock}
+        visibleColumns={visibleColumns}
+        nsParams={nsParams}
+        onSubmitSearch={onSubmitSearchMock}
+        onSelectRow={onSelectRowMock}
+        onCreate={onCreateMock}
+        onEdit={onEditMock}
+      />
+    </Router>
+  </Harness>
+);
+
+const renderSearchAndSort = props => {
+  return render(getSearchAndSortComponent(props));
+};
+
+describe('SearchAndSort component', () => {
+  afterEach(() => {
+    onSubmitSearchMock.mockClear();
+    onSelectRowMock.mockClear();
+    EditRecordComponentMock.mockClear();
+    onCreateMock.mockClear();
+    onEditMock.mockClear();
+  });
+
+  it('should be rendered', () => {
+    const { getByText } = renderSearchAndSort(searchAndSortProps({
+      parentResources: resources(1, false),
+      isFullScreen: true,
+      layer: '?layer=create',
+    }));
+
+    expect(getByText('Results Label')).toBeDefined();
+    expect(getByText('1 job profile')).toBeDefined();
```

Comment:
Can we use `toBeInTheDocument()` matcher instead of `toBeDefined()` to make intention clearer?

Reply:
Hi, Javokhir! Could you, please, explain the difference between `toBeInTheDocument()` and `toBeDefined()`. Why do you insist on it?

Reply:
Hi, @OleksandrHladchenko1 . With `toBeInTheDocument()`, we can make our intention which is checking if element in the document or not clearer and readible. `toBeDefined` is better to check if variable is not undefined as in the `Jest` [docs](https://jestjs.io/docs/expect#tobedefined) .